fix(textfield): Fix textarea label from overlapping border.

moog16 opened the pull request:
Removed textarea border to prevent double borders.
Added line-height to textarea label.

Fixes https://github.com/material-components/material-components-web/issues/1608
https://www.pivotaltracker.com/story/show/153034989

havenchyk commented on packages/mdc-textfield/mdc-text-field.scss:
```diff
@@ -445,7 +445,6 @@
   .mdc-text-field__input {
     padding: $padding-inset;
     padding-top: $padding-inset * 2;
-    border: 1px solid transparent;
```
I'm not 100% sure, but in active state textarea should have `2px` border and if you just remove this border, the whole border will become `1px`

![screencast 2017-12-08 15-39-06](https://user-images.githubusercontent.com/1291032/33766330-12557fd4-dc2e-11e7-8919-8850488151cb.gif)

havenchyk commented:
@moog16 can transparent background solve the initial problem instead of line-height? https://github.com/material-components/material-components-web/blob/master/packages/mdc-textfield/mdc-text-field.scss#L479

kfranqueiro commented:
I believe the reason for the non-transparent background is so that the label is still readable in cases where the text entered into the textarea overlaps it due to overflow/scrolling.

moog16 commented:
@

So a few things are being addressed in this PR:

1. I updated the label background to be transparent as opposed to white. This is what was causing the overlap with the border as well as some text in situations where zoom was used.
1. There was another issue when moving the fullwidth textarea handle horizontally, and the outer element would not collapse/expand with the textarea element. I fixed this with the `resize; vertical;` property.
1. The 2nd issue would also come up when it was not fullwidth. The textarea wrapper element needed the css property `width: fit-content`, which allows for the outer div to take the width of the children.
1. I also added line-height to the label, which was also used in normalize.css. There is a conflict with Safari's default <label /> line-height that caused the overlap in the top border.
